website/docs/registry/api.html.md

```diff
@@ -36,9 +36,150 @@ For example, if discovery produces the URL `https://modules.example.com/v1/`
 then this API would use full endpoint URLs like
 `https://modules.example.com/v1/{namespace}/{name}/{provider}/versions`.
 
-The example request URLs shown in this document are for the public
-[Terraform Registry](https://registry.terraform.io), and use its API base
-URL of `https://registry.terraform.io/v1/modules/` .
+## Base URL
+
+The example request URLs shown in this document are for the public [Terraform
+Registry](https://registry.terraform.io), and use its API `<base_url>` of
+`https://registry.terraform.io/v1/modules/`. Note that although the base URL in
+the [discovery document](#service-discovery) _may include_ a trailing slash, we
+include a slash after the placeholder in the `Path`s below for clarity.
+
+## List Modules
+
+These endpoints list modules according to some criteria.
+
+| Method | Path                                  | Produces                   |
+| ------ | ------------------------------------- | -------------------------- |
+| `GET`  | `<base_url>` (no trailing slash)      | `application/json`         |
```

Comment:
Is the `(no trailing slash)` necessary? It would work with both, right?

I fear this will make things more confusing than not.

Reply:
If you do have a slash we will 301 redirect so it does work for some value. But it feels odd to me that we would officially document a path that is a 301 redirect. This was the most succinct way I could find to not document the wrong thing without writing an essay about it and highlighting the confusion!

I considered changing the API to always have the slash but I think that just breaks conventions within the API itself for the sake of this one thing.

I don't mind removing this and people can just figure out for themselves to strip the slash if they want to avoid a pointless redirect but at least some clients wouldn't follow 301 by default (for example curl) so it looks kinda broken to document it as being the canonical endpoint?If you do have a slash we will 301 redirect so it does work for some value. But it feels odd to me that we would officially document a path that is a 301 redirect...

I considered changing the API to always have the slash but I think that just breaks conventions within the API itself for the sake of this one thing.

I don't mind removing this and people can just figure out for themselves to strip the slash if they want to avoid a pointless redirect but at least some clients wouldn't follow 301 by default (for example curl) so it looks kinda broken to document it as being the canonical endpoint?

Reply:
Right, my suggestion was maybe just to remove the `(no trailing slash)` text but continuing to document it as `<base_url>`. If you consider this a specification, its still exactly correct, we never say a trailing slash works, but we may avoid some confusion around the phrase.

I may be sensitive/pedantic here though around seeing that phrase and thinking "but why though"

Reply:
Oh and the other part here is that in fact the `base_url` came from discovery document and it _may_ have a trailing slash already - our public reg one does now although it would work to remove it.